numpy/core/src/multiarray/ctors.c

```diff
@@ -1983,15 +1981,25 @@ PyArray_FromInterface(PyObject *origin)
     /* Case for data access through pointer */
     if (attr && PyTuple_Check(attr)) {
         PyObject *dataptr;
+
         if (PyTuple_GET_SIZE(attr) != 2) {
             PyErr_SetString(PyExc_TypeError,
                     "__array_interface__ data must be a 2-tuple with "
                     "(data pointer integer, read-only flag)");
             goto fail;
         }
         dataptr = PyTuple_GET_ITEM(attr, 0);
-        if (PyString_Check(dataptr)) {
-            res = sscanf(PyString_AsString(dataptr),
+
+        /* Allow unicode type strings */
+        if (PyUnicode_Check(dataptr)) {
+            btmp = PyUnicode_AsASCIIString(dataptr);
+            if (btmp == NULL) {
+                goto fail;
+            }
+            dataptr = btmp;
+        }
```

Comment:
I don't think we should add this. This is for supporting `__array_interface__ = dict(data=("1234", True))` as an alternative spelling for `__array_interface__ = dict(data=(1234, True))`.

I can't think of a single reason this is a good idea, and we don't document it on https://numpy.org/doc/stable/reference/arrays.interface.html

Reply:
Ah, from the docs:

> ### Differences with Array interface (Version 2)
> The version 2 interface was very similar. The differences were largely aesthetic. In particular:
> * ...
> * The tuple returned from `__array_interface__[‘data’]` used to be a hex-string (now it is an integer or a long integer).
> * There was no `__array_interface__ `attribute

So it looks like this was intended to be removed.

Reply:
I'll make another PR to remove the string version of the data pointer, it wasn't present in v1.1.0, so appears to have been add later.

Reply:
OK, it was in v1.1.0, I was looking in the wrong place, but so was the integer option.

Reply:
At any rate it's documented as no longer supported since 2006, so I think we can quite safely deprecate it.